core-rust/mesh-api-server/src/mesh_api/server.rs

```diff
@@ -0,0 +1,178 @@
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+use crate::prelude::*;
+use std::future::Future;
+use std::sync::Arc;
+
+use super::metrics::MeshApiMetrics;
+use super::metrics_layer::MetricsLayer;
+use axum::extract::State;
+use axum::middleware::map_response;
+
+use axum::{
+    routing::{get, post},
+    Router,
+};
+
+use prometheus::Registry;
+use state_manager::state_manager::StateManager;
+use tower_http::catch_panic::CatchPanicLayer;
+
+use super::{handlers::*, ResponseError};
+
+use crate::mesh_api::{emit_error_response_event, InternalServerErrorResponseForPanic};
+
+#[derive(Clone)]
+pub struct MeshApiState {
+    pub network: NetworkDefinition,
+    pub state_manager: StateManager,
+}
+
+pub async fn create_server<F>(
+    bind_addr: &str,
+    shutdown_signal: F,
+    mesh_api_state: MeshApiState,
+    metric_registry: &Registry,
+) where
+    F: Future<Output = ()>,
+{
+    let router = Router::new()
+        .route("/network/status", post(handle_network_status))
+        .route("/network/list", post(handle_network_list))
+        .route("/network/options", post(handle_network_options))
+        .route("/account/balance", post(handle_account_balance))
+        // account/coins - not needed as we're not UTXO
+        .route("/block", post(handle_endpoint_todo))
+        .route("/block/transaction", post(handle_endpoint_todo))
+        .route("/call", post(handle_endpoint_not_supported))
+        .route("/construction/derive", post(handle_endpoint_todo))
+        .route("/construction/preprocess", post(handle_endpoint_todo))
+        .route("/construction/metadata", post(handle_endpoint_todo))
+        .route("/construction/payloads", post(handle_endpoint_todo))
+        .route("/construction/combine", post(handle_endpoint_todo))
+        .route("/construction/hash", post(handle_endpoint_todo))
+        .route("/construction/parse", post(handle_endpoint_todo))
+        .route("/construction/submit", post(handle_endpoint_todo))
+        // Below Indexer APIs - Not Required
+        .route("/search/transaction", post(handle_endpoint_not_supported))
+        .route("/events/blocks", post(handle_endpoint_not_supported))
+        // TODO mempool?
+        .route("/mempool", post(handle_endpoint_not_supported))
+        .route("/mempool/transaction", post(handle_endpoint_not_supported))
+        .with_state(mesh_api_state);
```

Comment:
Can we remove all the optional endpoints, per https://docs.cdp.coinbase.com/assets/images/mesh-api-quick-reference-guide-3f0d2011d255c271d294583df7ba9f4f.jpg?

And maybe reorder the entries accordingly?

Reply:
This is an example implementation for what's worth, https://github.com/near/nearcore/blob/ef032af3af713fb3957d6c270c1a6c7817ab211f/chain/rosetta-rpc/src/lib.rs#L872.

Reply:
Will move those optional at the end of the chain